File: Frontend/src/app/auth/interfaces/signin-methods.ts
export interface SignInMethods {
    hasEmail : boolean,
    hasPassword: boolean,
    externalSignMethods: ExternalSignInMethod[]
}

export interface ExternalSignInMethod {
    loginProvider : LoginProvider,
    providerName : string,
    identifier ?: string,
    configured: boolean,
}

export enum LoginProvider {
    Google = "google",
    GitHub = "gitHub"
}

export enum LoginPurpose {
    SignIn = "signIn",
    LinkAccount = "linkAccount"
}

File: Backend/markitAPI/markit.Application/Models/Authentication/SignInMethods.cs
using markit.Application.Models.Authentication.Enums;

namespace markit.Application.Models.Authentication
{
    public record SignInMethods(bool HasEmail, bool HasPassword, List<ExternalSignInMethod> ExternalSignMethods);
    public record ExternalSignInMethod(LoginProvider LoginProvider, string? Identifier, bool Configured);
}
